syft/workers/base.py

```diff
@@ -112,7 +112,14 @@ def __init__(
         self.id = id
         self.is_client_worker = is_client_worker
         self.log_msgs = log_msgs
-        self.verbose = verbose
+        if verbose is None:
+            self.verbose = hook.verbose if hasattr(hook, "verbose") else False
+        else:
+            self.verbose = verbose
+
+        if isinstance(hook, sy.TorchHook) and hasattr(hook, "_virtual_workers"):
+            hook._virtual_workers.add(self)
```

Comment:
This looks like it adds all workers to `hook._virtual_workers`. Either the name `_virtual_workers` or something about the logic doesn't quite make sense here. 🤔

Reply:
We keep track of the "virtual" workers (alice, bob, etc.) here. Hence the naming convention.

Reply:
This is in BaseWorker though, which means all workers get added to this dictionary.